### PR TITLE
Testing Infra Cleanup

### PR DESCRIPTION
**Pull request by mvpatel2000**
# What does this PR do?

Cleans up some test infra which will lay groundwork for migrating to nightly tests.

- Simplifies fixture files, consolidating into one file for autouse fixtures and one for the rest
- Removes sftp logging test for remote (expensive to run)
- Minor fixes throughout tests

Note: Tags reordered in workflow are so it makes it a little more readable with nightly tests. I removed that workflow and will merge separately because idt it will be completed before holidays.

# What issue(s) does this change relate to?

[CO-1520](https://mosaicml.atlassian.net/browse/CO-1520)

### CODE REVIEW

File: composer/trainer/trainer.py [hunk 1]
```diff
@@ -1675,6 +1675,7 @@ def fit(
 
         # Max Duration
         if duration is not None:
+            log.info(str(duration))
```
Comment (dakinggg) [inline composer/trainer/trainer.py:1678]: Is this log intentional? If so it should probably provide some more info about what it is logging

File: tests/conftest.py [hunk 2]
```diff
@@ -45,7 +41,7 @@ def _add_option(parser: pytest.Parser, name: str, help: str, choices: Optional[L
     )
 
 
-def _get_option(config: pytest.Config, name: str, default: Optional[str] = None) -> str:
+def _get_option(config: pytest.Config, name: str, default: Optional[str] = None) -> str:  # type: ignore
```
Comment (dakinggg) [inline tests/conftest.py:44]: Why was this type ignore added?

**Review by nik-mosaic — APPROVED** [review]

File: tests/fixtures/autouse_fixtures.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+
+@pytest.fixture(autouse=True)
+def set_loglevels():
+    """Ensures all log levels are set to DEBUG."""
+    logging.basicConfig()
+    logging.getLogger(composer.__name__).setLevel(logging.DEBUG)
+
+
+@pytest.fixture
```
Comment (dakinggg) [inline tests/fixtures/autouse_fixtures.py:78]: Is this supposed to be in the other fixtures file?

**Review by dakinggg — APPROVED** [review]
LGTM

File: tests/conftest.py (hunk 2 above)
Comment (mvpatel2000) [reply tests/conftest.py:44]: Function not used in file (imported elsewhere)

**Review by bandish-shah — APPROVED** [review]
👍

**Pull request merged**